tests/structuredheadings/matchheading.js

```diff
@@ -112,6 +112,46 @@
           "H6 changed back to P"
         );
 
+    },
+
+    "MatchHeading in list converts list level": function () {
+      this.editorBot.setHtmlWithSelection("<ul><li>^ListItem</li></ul>");
+      this.editorBot.execCommand("matchHeading");
+      var updatedContent = bender.tools.getHtmlWithSelection(this.editorBot.editor);
+
+      assert.areSame(
+            "<h1 class=\"autonumber autonumber-0\">^ListItem</h1>", updatedContent,
+            "LI changed to H1"
+          );
+
+    },
+
+    "MatchHeading in list following heading sets next level": function () {
+      this.editorBot.setHtmlWithSelection("<h1 class=\"autonumber autonumber-0\">Header</h1>" +
+       "<ul><li>^ListItem</li></ul>");
+      this.editorBot.execCommand("matchHeading");
+      var updatedContent = bender.tools.getHtmlWithSelection(this.editorBot.editor);
+
+      assert.areSame(
+            "<h1 class=\"autonumber autonumber-0\">Header</h1>" +
+            "<h2 class=\"autonumber autonumber-1\">^ListItem</h2>", updatedContent,
+            "LI cahanged to h2"
+          );
+
+    },
+
+    "MatchHeading in list keeps sublists": function () {
+      this.editorBot.setHtmlWithSelection("<ul><li>^ListItem" +
+        "<ul><li>SubListItem</li></ul></li></ul>");
+      this.editorBot.execCommand("matchHeading");
+      var updatedContent = bender.tools.getHtmlWithSelection(this.editorBot.editor);
+
+      assert.areSame(
+              "<h1 class=\"autonumber autonumber-0\">^ListItem</h1>" +
+              "<ul><li>SubListItem</li></ul>", updatedContent,
+              "Sublist in tact"
+            );
+
```

Comment:
we could probably use some tests with things like:

- a formatting element inside a list item
- more complex list item content 
- the really terrible list item that has text before and after an inner nested list (for ex `<li>foo<ul>...</ul>bar</li>` which ckeditor does poorly with sometimes